File: backend/src/Domain/Entities/Product.cs
namespace ProductionMove.Domain.Entities;

public class Customer
{
    public string Phone { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;
}

public class Product : BaseEntity
{
    public int Id { get; set; }

    public ProductStatus Status { get; set; }

    public DateTime DateOfManufacture { get; set; }

    public DateTime? SaleDate { get; set; } = null;

    public string ProductLineId { get; set; } = string.Empty;

    public string FactoryId { get; set; } = string.Empty;

    public string? DistributorId { get; set; } = null;

    public string? DistributionId { get; set; } = null;

    public virtual Customer? Customer { get; set; } = null;

    public virtual ICollection<Warranty> Warranties { get; set; } = new List<Warranty>();
}

File: frontend/src/data/entities/Product.ts
import { ProductStatus } from "../enums/ProductStatus"

export interface Customer {
    phone: string,
    name: string
}

export default interface Product {
    id: number,
    status: ProductStatus,      // Trạng thái hiện tại của sản phẩm
    dateOfManufacture: number,  // Ngày sản xuất
    saleDate: number|null,      // Ngày bán, null nếu chưa bán
    productLineId: string,
    factoryId: string,          // Id cơ sở sản xuất
    distributionId: string|null,// Đơn hàng nhập, null nếu sản phẩm chưa được nhập vẫn ở cơ sở sản xuất
    customer: Customer|null     // Thông tin khách hàng mua, null nếu sản phẩm chưa bán
}